Implement Wellington design

PeeterPaal opened the pull request:
Implements Wellington design, which is a minimal base layout for developing new sites. The layout includes most used components, a singular layout (Front page), and minimal JS/CSS files.

Compared to previous Voog designs, Bower and Grunt are not used. Instead, building and minifying JS/CSS/assets is done using Taskfile. All required packages will be installed via NPM.

For testing and initializing dev environment:

Firstly, clone repo and install packages by running `npm install`

Try different tasks provided in the Taskfile. These include:
- `npm run dev` – initializes watchers for JS/CSS/assets. Process these files automatically whenever something changes.
- `npm run dev:kit` – does the same as previous, but also initializes Voog Kit to automatically push changes to site defined in `.voog` config file.
- `npm run prod` – builds and minifies JS/CSS/assets

Check that content areas work without issues and layout is usable:
- navigation menu is present
- language menu is present
- buttons are _button-like_ (paddings and borders are present)

PeeterPaal committed:
Add base files

Add basic files for new layout to work. This includes components,
layouts, styles & scripts.

Only one layout is included, which includes main menu, language menu,
search and one content area.

PeeterPaal committed:
Use Taskfile instead of Makefile

PeeterPaal committed:
Make Taskfile compatible with Windows

PeeterPaal committed:
Use concatenated JS files for minification

PeeterPaal committed:
Show image title under image if present

PeeterPaal committed:
Update README

PeeterPaal committed:
Improve watching changed files

Watch changed assets and images. In addition, watch JS and CSS changes
in parallel to include all changes.

Voog Kit is now correctly initialized when running dev-kit task.

PeeterPaal committed:
Relocate `content-formatted` specific styles

PeeterPaal committed:
Update manifest

tanelj commented on components/html-head.tpl:
```diff
@@ -0,0 +1,44 @@
+  <link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-tools/latest/edicy-tools.css">
+{% endif %}
+
+<link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.css">
```
```suggestion
{% if site.search.enabled -%}
  {% stylesheet_link "libs/edicy-search/latest/edicy-search.css" static_host="true" %}
{%- endif %}
```

tanelj commented on components/html-head.tpl:
```diff
@@ -0,0 +1,44 @@
+
+{% if editmode %}
+  <link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-tools/latest/edicy-tools.css">
+{% endif %}
```
Let's use [stylesheet_link](https://www.voog.com/developers/markup/tags/stylesheet-link) tag.
```suggestion
{% if editmode -%}
  {% stylesheet_link "libs/edicy-tools/latest/edicy-tools.css" static_host="true" %}
{%- endif %}
```

tanelj commented on components/search.tpl:
```diff
@@ -0,0 +1,12 @@
+  <div class="search-inner js-modal">
+    <form id="search" class="search-form js-search-form" method="get" action="#">
+      <div class="search-left">
+        <input id="onpage_search" class="search-input js-search-input js-input-dynamic" type="text" name="" placeholder="{{  'search' | lc }}">
```
Let's add [escape_once](https://www.voog.com/developers/markup/basics/filters#escape_once) filter to ensure that external string doesn't brake html.
```suggestion
        <input id="onpage_search" class="search-input js-search-input js-input-dynamic" type="text" name="" placeholder="{{ "search" | lc | escape_once }}">
```

tanelj commented on components/search.tpl:
```diff
@@ -0,0 +1,12 @@
+        <input id="onpage_search" class="search-input js-search-input js-input-dynamic" type="text" name="" placeholder="{{  'search' | lc }}">
+      </div>
+      <div class="search-right">
+        <input class="search-submit js-search-submit" type="submit" value="Search" name="">
```
```suggestion
        <input class="search-submit js-search-submit" type="submit" value="{{ "search" | lc | escape_once }}" name="">
```

tanelj commented on components/template-meta.tpl:
```diff
@@ -0,0 +1,46 @@
+{%- endif -%}
+
+<meta property="og:type" content="{%- if article -%}article{%- else -%}website{%- endif -%}">
+<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first:"/" }}">
```
```suggestion
<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first: "/" }}">
```

tanelj commented on components/template-meta.tpl:
```diff
@@ -0,0 +1,46 @@
+<meta property="og:type" content="{%- if article -%}article{%- else -%}website{%- endif -%}">
+<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first:"/" }}">
+<meta property="og:title" content="{%- title -%}">
+<meta property="og:site_name" content="{{ page.site_title | escape }}">
```
[escape_once](https://www.voog.com/developers/markup/basics/filters#escape_once)` should be better.
```suggestion
<meta property="og:site_name" content="{{ page.site_title | escape_once }}">
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,14 @@
+{% sitejs_include %}
+
+{% comment %}Site search related javascript components.{% endcomment %}
+{% if site.search.enabled %}
```
```suggestion
{% if site.search.enabled -%}
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,14 @@
+{% if site.search.enabled %}
+  <script src="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.js"></script>
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', '{{ 'search_noresults' | lc | escape }}');</script>
+{% endif %}
```
```suggestion
{%- endif %}
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,14 @@
+{% comment %}Site search related javascript components.{% endcomment %}
+{% if site.search.enabled %}
+  <script src="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.js"></script>
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', '{{ 'search_noresults' | lc | escape }}');</script>
```
```suggestion
  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', {{ 'search_noresults' | lc | json }});</script>
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,14 @@
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', '{{ 'search_noresults' | lc | escape }}');</script>
+{% endif %}
+
+{% comment %}GOOGLE ANALYTICS INITIATION{% endcomment %}
```
```suggestion
{% comment %}SITE ANALYTICS INITIATION{% endcomment %}
```

tanelj commented on components/menu-main.tpl:
```diff
@@ -0,0 +1,31 @@
+          {% endfor %}
+
+          {% if editmode %}
+            <li>{% menuadd parent=item %}</li>
```
Second level "Add subpage"  button is not shown where the parent page is active, but has no children. It must be always shown for non root page.

tanelj commented on components/menu-main.tpl:
```diff
@@ -0,0 +1,31 @@
+<nav class="menu-main">
+  <ul class="menu">
```
White spaces removal can be improved.

tanelj commented on components/menu-lang.tpl:
```diff
@@ -0,0 +1,13 @@
+<div class="menu-lang">
+  {% for language in site.languages %}
```
```suggestion
  {% for language in site.languages -%}
```

tanelj commented on components/menu-lang.tpl:
```diff
@@ -0,0 +1,13 @@
+    <span>
+      <a class="lang-flag lang-flag-{{ language.code }}{% if language.selected? %} active{% endif %}" href="{{ language.url }}" data-lang-code="{{ language.locale }}">{{ language.title }}</a>
+    </span>
+  {% endfor %}
```
```suggestion
  {%- endfor %}
```

tanelj commented on components/menu-lang.tpl:
```diff
@@ -0,0 +1,13 @@
+    </span>
+  {% endfor %}
+
+  {% if editmode %}
```
```suggestion
  {% if editmode -%}
```

tanelj commented on components/menu-lang.tpl:
```diff
@@ -0,0 +1,13 @@
+    <span class="edit-btn">
+      {% languageadd %}
+    </span>
+  {% endif %}
```
```suggestion
  {%- endif %}
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,14 @@
+{% comment %}SITE WIDE JAVASCRIPTS{% endcomment %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
+<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
```
```suggestion
<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
{% if editmode -%}
  <script src="{{ javascripts_path }}/editmode.min.js?v={{ template_settings.version }}"></script>
{%- endif %}
```

tanelj commented on sources/stylesheets/_content-formatted.scss:
```diff
@@ -0,0 +1,16 @@
+    line-height: 1.3em;
+    content: attr(data-title);
+  }
+}
```
Please add image/map/video container related base styles: https://www.voog.com/developers/markup/basics/smarter-media

Please add form area basic CSS boilerplate (small/medium/large).

tanelj reviewed (changes requested):
Nice work!

Here is first feedback mostly about Liquid tweaks.

PeeterPaal committed:
Improve whitespaces & use more liquid tags

- Use stylesheet_link tags
- Include editmode JS
- Improve stripping whitespaces

PeeterPaal committed:
Improve main menu component

- Show submenu is subitem is selected
- Show hidden items for selected item
- Improve stripping whitespaces

t-p-developer commented on javascripts/editmode.js:
```diff
@@ -0,0 +1,383 @@
+              .removeClass('is-cropped')
+              .addClass('not-cropped')
+              .css('opacity', .1)
+              ;
```
nit! Let's put `;` in the above line. Please check the rest of the file also, there is the same inconsistency.

t-p-developer commented on javascripts/editmode.js:
```diff
@@ -0,0 +1,383 @@
+          itemData.get({
+            success: function (data) {
+              if (data.item_image) {
+                itemData.remove(itemImageKey, {
+                  success: function (data) {
+                    itemData.remove(cropStateKey, {
+                      success: function (data) {
+                        if (itemType !== 'article') {
+                          handleProductImage(placeholderText, itemId, null, $el);
+                        } else {
+                          addProductImagePlaceholder($el, placeholderText);
+                        }
+                      }
+                    });
+                  }
```
nit! Can we extract this logic? At the moment, it seems like a `callback-hell`...

PeeterPaal replied:
This logic is copied over from other standard layouts. I would keep it largely unchanged.

PeeterPaal replied:
Took a closer look – this part seems to be a bit unnecessary, as crop toggles are not present in every design. This logic also seems tied to product & article images, which went unnoticed.

I'd keep other general scripts (bgpicker, custom data etc.), as these are perhaps more widely used.

t-p-developer commented on javascripts/editmode.js:
```diff
@@ -0,0 +1,383 @@
+          if (dataEntity == 'site') {
+            var entityData = new Edicy.CustomData({
+              type: 'site'
+            });        } else {
```
nit! Formatting.

PeeterPaal committed:
Add styles for texteditor media

PeeterPaal committed:
Add basic styles for forms

PeeterPaal committed:
Remove image drop area scripts

PeeterPaal committed:
Add prettier config

PeeterPaal committed:
Format `editmode` & `application` JS w/ Prettier

t-p-developer commented on .prettierrc:
```diff
@@ -0,0 +1,8 @@
+{
```
🏅

t-p-developer commented on .prettierrc:
```diff
@@ -0,0 +1,8 @@
+{
+  "arrowParens": "avoid",
+  "bracketSameLine": false,
+  "bracketSpacing": false,
+  "printWidth": 100,
+  "singleQuote": true,
+  "trailingComma": "es5"
+}
```
🏅

tanelj reviewed (approved):
Nice!

PeeterPaal committed:
Update taskfile

Enable building design editor custom styles CSS to .tpl files, if
custom styles are present.

PeeterPaal committed:
Add bgpicker to front page layout

tanelj commented on components/bg-picker-scripts.tpl:
```diff
@@ -0,0 +1,30 @@
+<script type="text/javascript">
+  $(function() {
+    $('.bg-picker-area').each(function(index, pickerArea) {
+      var $picker = $(pickerArea).find('.bg-picker');
```
Its nice to guard against missing `bg-picker` element.

tanelj commented on components/template-styles.tpl:
```diff
@@ -0,0 +1,12 @@
+  }
+
+  .body-bg-image {
+    background-image: url("{{ body_bg_image }}");
```
NB! Do not render when `body_bg_image` is empty (renders current page html when url is missing).

```suggestion
    {% if  body_bg_image != blank%}background-image: url("{{ body_bg_image }}");{% endif %}

tanelj commented on components/header.tpl:
```diff
@@ -0,0 +1,11 @@
+  <div class="header-right">
+    {% include "menu-lang" %}
+
+    {% if site.search.enabled %}
```
```suggestion
    {%- if site.search.enabled %}
```

tanelj commented on components/header.tpl:
```diff
@@ -0,0 +1,11 @@
+
+    {% if site.search.enabled %}
+      {% include "search" %}
+    {% endif %}
```
```suggestion
    {%- endif %}
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,36 @@
+{% comment %}SITE ANALYTICS INITIATION{% endcomment %}
+{% unless editmode %}{{ site.analytics }}{% endunless %}
+
+{% if editmode %}
```
```suggestion
{%- if editmode %}
```

tanelj commented on components/javascripts.tpl:
```diff
@@ -0,0 +1,36 @@
+    </script>
+    {%- include "bg-picker-scripts" -%}
+  {% endeditorjsblock %}
+{% endif %}
```
```suggestion
{%- endif %}
```

tanelj commented on components/template-meta.tpl:
```diff
@@ -0,0 +1,48 @@
+{%- endif -%}
+
+{%- if og_image -%}
+  {%- comment -%}"http:" and "https:" strings are removed and readded to ensure that older bg-picker images will have protocol.{%- endcomment -%}
+  {%- if og_image.url -%}<meta property="og:image" content="{{ og_image.url | replace_first: "http:", "" | replace_first: "https:", "" | prepend: "https:" }}">{%- endif -%}
```
The `og_image` is always Image object (is not coming from object data attribute) so:
```suggestion
  {{% if og_image.url %}<meta property="og:image" content="{{ og_image.url }}">{% endif %}
```

tanelj commented on layouts/front_page.tpl:
```diff
@@ -0,0 +1,42 @@
+    <div class="body-bg-image js-background-image"></div>
+    <div class="body-bg-color js-background-color"></div>
+
+    {% if editmode %}
```
```suggestion
    {%- if editmode %}
```

tanelj commented on layouts/front_page.tpl:
```diff
@@ -0,0 +1,42 @@
+        data-bg-color="{{ body_bg_color }}"
+        data-bg-image-sizes="{{ body_bg_image_sizes_str | escape }}"
+      ></button>
+    {% endif %}
```
```suggestion
    {% endif -%}
```

tanelj commented on manifest.json:
```diff
@@ -0,0 +1,144 @@
+    {
+      "content_type": "page",
+      "component": false,
+      "file": "layouts/front_page.tpl",
+      "layout_name": "page_default",
+      "title": "Front page"
```
Maybe we should name it common page?

```suggestion
      "file": "layouts/common_page.tpl",
      "layout_name": "page_default",
      "title": "Common page"
```

tanelj reviewed (changes requested):
Nice work! Here are some feedback.

PeeterPaal committed:
Improve stripping whitespace

PeeterPaal committed:
Improve bgpicker & og image handling

- Do not render background image style if not present
- Do not initialize bgpicker if no element is found
- Simplify og image url rendering

PeeterPaal committed:
Rename page layout to `Common page`

PeeterPaal committed:
Fix bg color & image placement

tanelj reviewed (approved):
Nice work!
